.github/actions/generate-metadata/action.yml

```diff
@@ -0,0 +1,54 @@
+---
+name: Build environment
+description: Create build environment
+
+inputs:
+    gh_token:
+        description: GitHub token
+        required: true
+        type: string
+    cache:
+        description: use caching or not
+        required: true
+        type: boolean
+
+runs:
+    using: composite
+    steps:
+        - name: Restore cached Plugins
+          id: cache-plugins-restore
+          uses: actions/cache/restore@v3
+          with:
+            path: plugins_metadata.json
+            key: plugins_metadata
+        - name: debug
+          run: |
+            echo ${{ steps.cache-plugins-restore.outputs.cache-hit }}
+            echo ${{ inputs.cache }}
+          shell: bash
+
+        - name: fetch metadata
+          if: '!(inputs.cache || steps.cache-plugins-restore.outputs.cache-hit != 'true')'
```

Comment:
```suggestion
          # the expression is complex because of https://github.com/actions/runner/issues/753
          if: (!(inputs.cache || steps.cache-plugins-restore.outputs.cache-hit != 'true'))
```